src/components/dataList/dataListFilter/dataListFilter.tsx

```diff
@@ -0,0 +1,162 @@
+                    <AvatarIcon
+                        icon={IconType.SEARCH}
+                        size="md"
+                        className="shrink-0"
+                        variant={isFocused ? 'primary' : 'neutral'}
+                    />
+                )}
+                {state === 'loading' && <Spinner size="lg" variant="primary" className="m-1 shrink-0" />}
+                <input
+                    type="search"
+                    className={classNames(
+                        'size-full truncate bg-transparent pl-3 caret-neutral-500 outline-none search-cancel:appearance-none',
+                        'placeholder:text-base placeholder:font-normal placeholder:leading-tight placeholder:text-neutral-300',
+                    )}
+                    onFocus={handleInputFocus}
+                    onBlur={handleInputBlur}
+                    onChange={handleSearchValueChange}
+                    placeholder={placeholder}
+                    value={searchValue}
+                />
+                {displayClearIcon && (
+                    <button className="mr-2 p-2.5 md:mr-4" onClick={handleClear}>
+                        <Icon icon={IconType.CLOSE} size="sm" className="text-neutral-300" />
+                    </button>
+                )}
+                <div className="flex flex-row gap-2 md:gap-3">
+                    {onFilterClick && (
+                        <>
+                            <Button
+                                iconLeft={IconType.FILTER}
+                                variant="tertiary"
+                                size="md"
+                                onClick={onFilterClick}
+                                className="hidden md:flex"
+                            >
+                                {/* TODO: apply internationalisation to Filter label [APP-2627] */}
+                                Filter
+                            </Button>
+                            <Button
+                                iconLeft={IconType.FILTER}
+                                variant="tertiary"
+                                size="md"
+                                onClick={onFilterClick}
+                                className="md:hidden"
+                            />
+                        </>
+                    )}
+                    <DataListFilterSort
+                        activeSort={activeSort}
+                        sortItems={sortItems}
+                        onSortChange={onSortChange}
+                        triggerClassNames="hidden md:flex"
+                        /* TODO: apply internationalisation to Sort label [APP-2627] */
+                        triggerLabel="Sort"
+                    />
+                    <DataListFilterSort
+                        activeSort={activeSort}
+                        sortItems={sortItems}
+                        onSortChange={onSortChange}
+                        triggerClassNames="md:hidden"
```

Comment:
It's like the button component, I tried to update the `DataListFilterSort` component to render two dropdowns, one with the sort label and the other one without, but it was then a bit weird and I thought that the actual styling should come from the parent component

Reply:
Wait, why two dropdowns and not two trigger buttons? The reason I am pro-handling this inside is that this is the only place where this component is being used; it's also not exported from the library.

Reply:
Mm.. I was sure radix-ui dropdown would throw an error in case of two buttons as dropdown trigger, but it doesn't. Regardless, the dropdown does not work if we pass two buttons 😅